src/reducers/existingThoughtMove.js

```diff
@@ -58,23 +61,45 @@ export default (state, { oldPath, newPath }) => {
   // if the contexts have changed, remove the value from the old contextIndex and add it to the new
   const subthoughtsOld = (state.contextIndex[contextEncodedOld] || [])
     .filter(child => !equalThoughtRanked(child, { value, rank: oldRank }))
-  const subthoughtsNew = (state.contextIndex[contextEncodedNew] || [])
-    .filter(child => !equalThoughtRanked(child, { value, rank: oldRank }))
-    .concat({
-      value,
-      rank: newRank,
-      lastUpdated: timestamp()
-    })
-
-  const recursiveUpdates = (thoughtsRanked, contextRecursive = [], accumRecursive = {}) => {
-
-    return getThoughtsRanked(thoughtsRanked, state.thoughtIndex, state.contextIndex).reduce((accum, child) => {
+  const subthoughtsNew = sameContext ?
+    (state.contextIndex[contextEncodedNew] || [])
+      .filter(child => !equalThoughtRanked(child, { value, rank: oldRank }))
+      .concat({
+        value,
+        rank: newRank,
+        lastUpdated: timestamp()
+      })
+    :
+    _.uniqBy((state.contextIndex[contextEncodedNew] || [])
+      .concat({
+        value,
+        rank: newRank,
+        lastUpdated: timestamp()
+      }),
```

Comment:
Notice how `state.contextIndex[contextEncodedNew] || []` and:

```js
  .concat({
    value,
    rank: newRank,
    lastUpdated: timestamp()
  })
```

are duplicated, obscuring the precise difference between the two different conditions. See if you can rewrite this without duplication to make the logic clearer.